Lib/logging/__init__.py

```diff
@@ -153,14 +153,14 @@ def addLevelName(level, levelName):
         _releaseLock()
 
 if hasattr(sys, '_getframe'):
-    currentframe = lambda: sys._getframe(3)
+    currentframe = lambda level: sys._getframe(level)
 else: #pragma: no cover
-    def currentframe():
+    def currentframe(level):
         """Return the frame object for the caller's stack frame."""
         try:
             raise Exception
         except Exception:
-            return sys.exc_info()[2].tb_frame.f_back
+            return sys.exc_info()[level-1].tb_frame.f_back
```

Comment:
I believe this is incorrect. `exc_info` returns three items and probably all we can do is
```suggestion
            frame = sys.exc_info()[2].tb_frame.f_back
        while frame and stacklevel > 0:
            frame = frame.f_back
            stacklevel -= 1
        if not frame:
            raise ValueError("call stack is not deep enough")
        return frame
```